nitro-shim/tools/gvproxy/gvproxy.go

```diff
@@ -0,0 +1,492 @@
+// The code below is taken (almost) verbatim from here:
+// https://github.com/containers/gvisor-tap-vsock/blob/a39274e8244767c19c74fa68e9d1d8316cf9e175/cmd/gvproxy/main.go
+// The only modification we made was to remove the forward-SSH-by-default rule.
+//
+// In the above revision, gvisor-tap-vsock is licensed under the Apache 2.0
+// license, which permits us to use the code here.
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"net"
+	"net/http"
+	"net/http/pprof"
+	"net/url"
+	"os"
+	"os/signal"
+	"runtime"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/containers/gvisor-tap-vsock/pkg/net/stdio"
+	"github.com/containers/gvisor-tap-vsock/pkg/sshclient"
+	"github.com/containers/gvisor-tap-vsock/pkg/transport"
+	"github.com/containers/gvisor-tap-vsock/pkg/types"
+	"github.com/containers/gvisor-tap-vsock/pkg/virtualnetwork"
+	"github.com/dustin/go-humanize"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+var (
+	debug           bool
+	mtu             int
+	endpoints       arrayFlags
+	vpnkitSocket    string
+	qemuSocket      string
+	bessSocket      string
+	stdioSocket     string
+	forwardSocket   arrayFlags
+	forwardDest     arrayFlags
+	forwardUser     arrayFlags
+	forwardIdentify arrayFlags
+	sshPort         int
+	pidFile         string
+	exitCode        int
+)
+
+const (
+	gatewayIP   = "192.168.127.1"
+	sshHostPort = "192.168.127.2:22"
+)
+
+func main() {
```

Comment:
We could probably drop all the sockets we don't actually intend to use here, but otherwise LGTM